src/main/java/org/openlmis/AuditLogInitializer.java

```diff
@@ -58,29 +65,48 @@ public void run(String... args) {
 
     //For each one...
     for (Object object : repositoryMap.values()) {
+      if (object instanceof PagingAndSortingRepository) {
+        createSnapshots((PagingAndSortingRepository<?, ?>) object);
+      } else if (object instanceof CrudRepository) {
+        createSnapshots((CrudRepository<?, ?>) object);
+      }
+    }
+  }
 
-      //... retrieve all of its domain objects and...
-      Iterable<?> domainObjects = ((CrudRepository)(object)).findAll();
-
-      //... for each one...
-      for (Object o : domainObjects) {
+  private void createSnapshots(PagingAndSortingRepository<?, ?> repository) {
+    Pageable pageable = new PageRequest(DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE);
 
-        //...check whether there exists a snapshot for it in the audit log.
-        // Note that we don't care about checking for logged changes, per se,
-        // and thus use findSnapshots() rather than findChanges()
-        BaseEntity baseEntity = (BaseEntity)o;
+    while (true) {
+      Page<?> page = repository.findAll(pageable);
 
-        QueryBuilder jqlQuery = QueryBuilder.byInstanceId(baseEntity.getId(), o.getClass());
-        List<CdoSnapshot> snapshots = javers.findSnapshots(jqlQuery.build());
+      if (!page.hasContent()) {
+        break;
+      }
 
-        //If there are no snapshots of the domain object, then take one
-        if (snapshots.size() == 0) {
-          javers.commit("System: AuditLogInitializer" , baseEntity);
-        }
+      page.forEach(this::createSnapshot);
 
-      }
+      pageable = pageable.next();
     }
+  }
+
+  private void createSnapshots(CrudRepository<?, ?> repository) {
```

Comment:
TBH this will help, but when I created the ticket I was thinking more about something that would not require us going through each requisition. Do you know how long this takes for Malawi before and after?